### DISCUSSION TITLE
Streaming a reqwest response as attachment

### DISCUSSION THREAD

**Discussion by sztomi** [rwf2/Rocket#1902] [Questions] [answered]
I'm looking for a way to stream (forward) a streamed response from reqwest to a client. To illustrate, here is my non-working code:

```rs
#[get("/github/<org>/<repo>/<sha>")]
async fn grab(org: String, repo: String, sha: String) -> ByteStream![Vec<u8>] {
  if let Ok(res) = reqwest::get(format!("https://github.com/{}/{}/archive/{}.tar.gz", org, repo, sha)).await {
    res.bytes_stream()
  }
}
```

I'm confused how I'm supposed to use the ByteStream type and macro. Is this even the right stream type?

**Comment by jebrosen** [comment, marked as answer]
`res.bytes_stream()` [returns](https://docs.rs/reqwest/0.11.5/reqwest/struct.Response.html#method.bytes_stream) a `Stream` of `Bytes`, which can be used with `ByteStream` in this or similar fashion:

```rust
#[get("/github/<org>/<repo>/<sha>")]
async fn grab(org: String, repo: String, sha: String) -> ByteStream![Bytes] {
  if let Ok(res) = reqwest::get(format!("https://github.com/{}/{}/archive/{}.tar.gz", org, repo, sha)).await {
    ByteStream(res.bytes_stream())
  } else {
    // TODO: handle error case
  }
}
```

(You may need to add `bytes` to your own `Cargo.toml` to name the `Bytes` type, since `reqwest` does not reexport it. `reqwest` 0.11 uses `bytes` 1.0.)

**Comment by thor314** [comment]
For anyone else who lands here looking for examples of streamed messages with reqwest, I wrote an example demo here, using EventStream.

https://github.com/thor314/rocket-examples/tree/main/streaming-reqwest-client-server

**Reply by sztomi** [reply]
Thanks!
